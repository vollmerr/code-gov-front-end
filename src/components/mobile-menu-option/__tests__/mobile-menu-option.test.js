import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux'

import MobileMenuOption from 'components/mobile-menu-option'

import { shallow, mount } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

describe('Testing <MobileMenuOption /> component', () => {
  const props = {
    expandedMobileMenuOptions: [],
    menuOption: {name: "test-name", url: "https://test-url.com"},
    hideMobileMenu: jest.fn(),
    toggleMobileMenuOption: jest.fn()
  };
  const store = mockStore(props);
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MobileMenuOption store={store} {...props} />);
  });
  it('renders Menu', () => {
    wrapper = wrapper.dive();
    expect(wrapper.exists()).toEqual(true);
  });
  it('renders Menu with multiple Links', () => {
    const props = {
      expandedMobileMenuOptions: [],
      menuOption: {
        links: [
          {name: "test-name", url: "https://test-url.com"},
          {name: "test-name2", url: "/test-url.com2"}
        ]},
      hideMobileMenu: jest.fn(),
      toggleMobileMenuOption: jest.fn()
    };
    wrapper = shallow(<MobileMenuOption store={store} {...props} />);
    wrapper = wrapper.dive();
    expect(wrapper.exists()).toEqual(true);
  });
  it('renders Menu with no Links', () => {
    const props = {
      expandedMobileMenuOptions: [],
      menuOption: {},
      hideMobileMenu: jest.fn(),
      toggleMobileMenuOption: jest.fn()
    };
    wrapper = shallow(<MobileMenuOption store={store} {...props} />);
    wrapper = wrapper.dive();
    expect(wrapper.exists()).toEqual(true);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
