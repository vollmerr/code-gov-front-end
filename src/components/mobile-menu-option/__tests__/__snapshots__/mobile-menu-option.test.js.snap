// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MobileMenuOption /> component should render correctly 1`] = `
<MobileMenuOption
  expandedMobileMenuOptions={Array []}
  hideMobileMenu={[Function]}
  menuOption={
    Object {
      "name": "test-name",
      "url": "https://test-url.com",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  toggleMobileMenuOption={[Function]}
/>
`;
